eth/filters/bor_filter.go

```diff
@@ -0,0 +1,154 @@
+func NewBorBlockLogsFilter(backend Backend, block common.Hash, addresses []common.Address, topics [][]common.Hash) *BorBlockLogsFilter {
+	// Create a generic filter and convert it into a block filter
+	filter := newBorBlockLogsFilter(backend, addresses, topics)
+	filter.block = block
+	return filter
+}
+
+// newBorBlockLogsFilter creates a generic filter that can either filter based on a block hash,
+// or based on range queries. The search criteria needs to be explicitly set.
+func newBorBlockLogsFilter(backend Backend, addresses []common.Address, topics [][]common.Hash) *BorBlockLogsFilter {
+	return &BorBlockLogsFilter{
+		backend:   backend,
+		addresses: addresses,
+		topics:    topics,
+		db:        backend.ChainDb(),
+	}
+}
+
+// Logs searches the blockchain for matching log entries, returning all from the
+// first block that contains matches, updating the start of the filter accordingly.
+func (f *BorBlockLogsFilter) Logs(ctx context.Context) ([]*types.Log, error) {
+	// If we're doing singleton block filtering, execute and return
+	if f.block != (common.Hash{}) {
+		receipt, err := f.backend.GetBorBlockReceipt(ctx, f.block)
+		if err != nil {
+			return nil, err
+		}
+
+		if receipt == nil {
+			return nil, errors.New("unknown block")
+		}
+		return f.borBlockLogs(ctx, receipt)
+	}
+
+	// Figure out the limits of the filter range
+	header, _ := f.backend.HeaderByNumber(ctx, rpc.LatestBlockNumber)
+	if header == nil {
+		return nil, nil
+	}
+	head := header.Number.Uint64()
+
+	if f.begin == -1 {
+		f.begin = int64(head)
+	}
+
+	// adjust begin for sprint
+	f.begin = nextSprintEnd(f.begin)
+
+	end := f.end
+	if f.end == -1 {
+		end = int64(head)
+	}
+
+	// Gather all indexed logs, and finish with non indexed ones
+	return f.unindexedLogs(ctx, uint64(end))
+}
+
+// unindexedLogs returns the logs matching the filter criteria based on raw block
+// iteration and bloom matching.
+func (f *BorBlockLogsFilter) unindexedLogs(ctx context.Context, end uint64) ([]*types.Log, error) {
+	var logs []*types.Log
+
+	for ; f.begin <= int64(end); f.begin = f.begin + 64 {
+		header, err := f.backend.HeaderByNumber(ctx, rpc.BlockNumber(f.begin))
+		if header == nil || err != nil {
+			return logs, err
+		}
+
+		// get bor block receipt
+		receipt, err := f.backend.GetBorBlockReceipt(ctx, header.Hash())
+		if receipt == nil || err != nil {
+			continue
+		}
+
+		// filter bor block logs
+		found, err := f.borBlockLogs(ctx, receipt)
+		if err != nil {
+			return logs, err
+		}
+		logs = append(logs, found...)
+	}
+	return logs, nil
+}
+
+// borBlockLogs returns the logs matching the filter criteria within a single block.
+func (f *BorBlockLogsFilter) borBlockLogs(ctx context.Context, receipt *types.BorReceipt) (logs []*types.Log, err error) {
+	if bloomFilter(receipt.Bloom, f.addresses, f.topics) {
+		found := filterLogs(receipt.Logs, nil, nil, f.addresses, f.topics)
+		logs = append(logs, found...)
```

Comment:
This append here is unnecessary. Especially when it is followed by another append on the line 133.

Reply:
Ah! got it. Thanks for catching it.

Reply:
Fixed.